pandas/tests/plotting/test_misc.py

```diff
@@ -277,6 +274,24 @@ def test_radviz(self):
         handles, labels = ax.get_legend_handles_labels()
         self._check_colors(handles, facecolors=colors)
 
+    @slow
+    def test_subplot_titles(self):
+        df = self.iris.drop('Name', axis=1).head()
+        # Use the column names as the subplot titles
+        title = list(df.columns)
+
+        # Case len(title) == len(df)
+        plot = df.plot(subplots=True, title=title)
+        self.assertEqual([p.title._text for p in plot], title)
+
+        # Case len(title) > len(df)
+        plot = df.plot(subplots=True, title=title + ['Ignore me!'])
```

Comment:
I think we should raise if length of passed list and number of axes are different.